packages/js/components/src/experimental-select-control/types.ts

```diff
@@ -32,16 +32,22 @@ export type getMenuPropsType = (
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
 ) => any;
 
-export type ChildrenProps = {
+export type ChildrenProps< ItemType > = {
 	items: ItemType[];
 	isOpen: boolean;
 	highlightedIndex: number;
-	getItemProps: getItemPropsType;
+	getItemProps: getItemPropsType< ItemType >;
 	getMenuProps: getMenuPropsType;
+	getItemLabel: getItemLabelType< ItemType >;
+	getItemValue: getItemValueType< ItemType >;
 };
 
-export type ChildrenType = ( {
+export type ChildrenType< ItemType > = ( {
 	items,
 	isOpen,
 	highlightedIndex,
-}: ChildrenProps ) => ReactElement | Component;
+}: ChildrenProps< ItemType > ) => ReactElement | Component;
+
+export type getItemLabelType< ItemType > = ( item: ItemType | null ) => string;
+
+export type getItemValueType< ItemType > = ( item: ItemType | null ) => string;
```

Comment:
Could we also allow the return of a `number` -> `string | number` for the value type? This is not a must if it will require a bunch of checks and conversions everywhere else, but if we can do it somewhat seamless then that would be really nice.